codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -272,10 +272,10 @@ private boolean allowNotificationEvent() {
         public ChannelPromise setSuccess(Void result) {
             if (allowNotificationEvent()) {
                 ++successfulCount;
-                if (successfulCount == expectedCount && doneAllocating) {
-                    promise.setSuccess(result);
-                    return super.setSuccess(result);
-                }
+            }
+            if (successfulCount == expectedCount && doneAllocating && !isDone()) {
```

Comment:
I understand consolidating the promise logic but I think my preference is to keep the two entry points as they represent logically different operations.  I think the bug here was that we were ultimately calling `this.setSuccess(null)` instead of `super.setSuccess(null)`. The usage pattern for `SimpleChannelPromiseAggregator` is to call `set/trySuccess(..)` or `set/tryFailure(..)` as many times as you want (which increments counts and keeps track of things) until `doneAllocatingPromises()` is called...I don't think `doneAllocatingPromises()` needs to call `this.setSuccess(..)` and introduce additional conditional logic for every `setSuccess(..)` call.